### ISSUE TITLE
Move code to GitLab or Codeberg

### ISSUE THREAD

**Issue by julianfairfax** [organicmaps/organicmaps#7059]
**Is your feature request related to a problem? Please describe.**
GitHub is not an open source platform, and being owned by Microsoft makes it more of a danger to open source projects, rather than a help.

**Describe the ideal solution**
It would be nice if the project's code were moved to GitLab or Codeberg.

**Comment by pastk** [comment]
I've heard that e.g. Codeberg's Terms of Service are not much better than GH's (i.e. Codeberg's platform code is open which is great, but the public service (SaaS) could a different story), would be great if someone could provide more details.

In any case moving to a different hosting is a lot of work, re-configuring of workflows and adjustment for contributors...

Adding an automatically synced mirror is a good start! Feel free to do it!

**Comment by pastk** [comment]
I see mirrors as a kind of quickly accessible backup mostly. And people can use them to browse the code, commits history etc. w/o using GH. In this sense it doesn't really matter who've created them, right?

Or do mirrors have more functionality indeed with issues mirroring etc.? Ref: https://github.com/orgs/FossifyOrg/discussions/37#discussioncomment-7779090

**Comment by pastk** [comment]
For reference:
https://gitlab.com/fdroid/admin/-/issues/159
A discussion of F-Droid moving from gitlab.com somewhere else (probably to a self-hosted gitlab).

**Comment by deevroman** [comment]
FYI: Codeberg, sr.ht can't cope with DDOS attack for the second day:
 https://social.anoxinon.de/@Codeberg/111740075240915378
https://fosstodon.org/@drewdevault/111739063243946284

This is another confirmation of why you should move development from GitHub.

**Comment by licaon-kter** [comment]
> This is another confirmation of why you shouldn't move development from GitHub.

Both large competitors to Github hit by DDOS at the same time...wow such coincidence... one that benefits who? I know.. let's stay on Github... unrelatedly.... :roll_eyes:

**Comment by deevroman** [comment]
Is it profitable for Gitlab? :))

I mean, it's easier for such a massive project as Organic Maps to stay on GitHub, which can provide it with the necessary capacities for free. What is the benefit of migrating to another platform? What risks can this eliminate?

**Comment by licaon-kter** [comment]
@deevroman well, Gitlab is under CF protection which... https://fosstodon.org/@arch@floofy.tech/111739294821803544 :roll-more-eyes:

**Comment by Julian-1-2-3-4-5** [comment]
Okay, technically [radicle](https://radicle.xyz) would be the ideal platform to move to, because its 100% decentralized and distributed, but i understand that it's a lot of work and might discourage some potential contributers, so it's a matter of what currently matters more, and should maybe be weighed against each other from time to time.

**Comment by pastk** [comment]
Recently I've researched a lot about Forgejo (the forge software which Codeberg runs) and I'm really inspired by their stance for software freedom.

The fact that Codeberg switched to Forgejo (from Gitea which became partly commercialized) and supports its development gives its a lot of points in my eyes!

**Comment by Ralimbahere** [comment]
We need to use more open source in our lives.

I support the Codeberg idea!

**Comment by biodranik** [comment]
Because it's way easier to contribute here, and it has a better infrastructure for CI and other related things. Cons from moving somewhere else now greatly overweight the pros.

**Comment by qgustavor** [comment]
I agree, and I asked because there wasn't an official announcement (at least I could not find one). To be fair, the only reason for me supporting moving from GitHub was the current state of USA's politics and its implications for GitHub, but I agree that's not a huge pro to support moving somewhere. Anyway, I hope other people with the same curiosity might find this thread.

**Comment by Misalf-git** [comment]
Seems to me the lack of transparency led to the inception of  https://codeberg.org/comaps  (by some of OM's previously most active contributers/members), while here on GitHub taciturnity continues.

**Comment by biodranik** [comment]
GitHub temporarily blocked the repo because a contributor who visited sanctioned territories was made an owner of the OM GitHub organization. That won't happen anymore.

**Comment by zyphlar** [comment]
> **Edit:** by the way, despite having a Telegram account and a bot, I don't use it and I avoid using it. Like people want to make other people move from GitHub, I want to see people moving away from Telegram to other places where E2EE is not only default, but enforced. Or that they make everything there E2EE, no exceptions. I had to implement E2EE in my Telegram bot myself because they don't support it in any sense. At their current state, it's a privacy nightmare and a bad option for people wishing privacy, like many Organic Maps users.

The relevant stuff is bridged to Matrix as well 💔

**Comment by biodranik** [comment]
Reasons to stay with GitHub: the largest open-source contributor network, familiar PR & issue workflow, Actions CI, broad integrations, zero infra to maintain, and easy onboarding/discoverability. This lets us focus on improving the app instead of running and maintaining servers. Development time is the most precious resource nowadays, and most of our users don't care where the code is hosted, but care about the app functionality and usability.

Cons of moving away from GitHub: additional efforts for everyone, additional spending of time and money on support of something that is already available, slowing down release cycles and new features delivery, increasing entry barriers for new contributors.

We can consider alternatives later, now our main focus is to _develop the best maps app_ and _deliver to users what they need_ sooner rather than later.

**Comment by qgustavor** [comment]
Issue report: biased reply. 

Expected output:

> "Reasons to stay with GitHub:"
> "Cons of staying on GitHub:"

Actual output:

> "Reasons to stay with GitHub:"
> "Cons of moving away from GitHub:"

Possible solution:

1. Any contributor answering with a proper non-biased answer with the pros and cons of staying and the pros and cons moving away 
2. Anyone following this issue waiting a discussion resulting in "moving away" as the chosen solution might look into other projects and consider this discussion done.

I will move away: while I showed support for staying on GitHub not only here but on my other projects, this answer is something I expected to find in a comedy show (like "say 'yes!' those who want me as the class leader, say 'no!' those who don't want him as the class leader" I guess I saw on some school-based comedy show), not on a serious GitHub discussion, so I'm unsubscribing from this issue and moving away.

**Comment by Jean-BaptisteC** [comment]
Today, a lot of comments in [issues](https://github.com/organicmaps/organicmaps/issues?q=is%3Aissue%20state%3Aclosed%20sort%3Aupdated-desc) has been updated and some issues has been locked by a main contributor of Organic Maps

It's a violation of freedom of expression!
How the project can be FOSS with this attitude.
